File: src/lib/ui/animations.ts
/**
 * Refined spring animation configurations for Framer Motion.
 *
 * Higher damping, weighted feel — not bouncy. Matches iOS 26 Liquid Glass motion.
 */

import type { Transition, Variants } from "framer-motion";

export const springTransitions = {
  /** Buttons, quick feedback. */
  snappy: {
    type: "spring",
    stiffness: 400,
    damping: 38,
  } as Transition,

  /** Content transitions, general movement. */
  smooth: {
    type: "spring",
    stiffness: 200,
    damping: 30,
  } as Transition,

  /** Entries, reveals, fade-ins. */
  gentle: {
    type: "spring",
    stiffness: 150,
    damping: 28,
  } as Transition,

  /** Sheet transitions, liquid glass morphing. */
  liquid: {
    type: "spring",
    stiffness: 180,
    damping: 28,
    mass: 1.1,
  } as Transition,

  /** Micro-interactions, immediate response. */
  quick: {
    type: "spring",
    stiffness: 500,
    damping: 40,
  } as Transition,

  /** Floating elements entering view. */
  floatingEntry: {
    type: "spring",
    stiffness: 250,
    damping: 32,
    mass: 0.8,
  } as Transition,

  /** Tab swipe / horizontal content transitions. */
  tabSwipe: {
    type: "spring",
    stiffness: 300,
    damping: 34,
  } as Transition,
} as const;

export const overlayVariants: Variants = {
  hidden: { opacity: 0 },
  visible: { opacity: 1, transition: { duration: 0.25 } },
  exit: { opacity: 0, transition: { duration: 0.2 } },
};

export const notificationVariants: Variants = {
  initial: {
    y: 80,
    opacity: 0,
  },
  animate: {
    y: 0,
    opacity: 1,
    transition: springTransitions.smooth,
  },
  exit: {
    y: 80,
    opacity: 0,
    transition: springTransitions.snappy,
  },
};

export const cardVariants: Variants = {
  idle: {
    scale: 1,
    boxShadow: "0 4px 12px rgba(0, 0, 0, 0.06)",
  },
  pressed: {
    scale: 0.985,
    boxShadow: "0 2px 8px rgba(0, 0, 0, 0.08)",
    transition: springTransitions.quick,
  },
  hover: {
    scale: 1.005,
    boxShadow: "0 8px 24px rgba(0, 0, 0, 0.08)",
    transition: springTransitions.snappy,
  },
};

export const searchResultVariants: Variants = {
  hidden: { opacity: 0 },
  visible: (i: number) => ({
    opacity: 1,
    transition: { duration: 0.25, delay: Math.min(i, 8) * 0.04 },
  }),
  exit: { opacity: 0, transition: { duration: 0.15 } },
};

/** Sheet corner radius values for progressive morphing. */
export const sheetCornerRadius = {
  floating: 28,
  mid: 20,
  full: 0,
} as const;

File: flutter/lib/ui/animations.dart
import 'package:flutter/physics.dart';

/// Spring physics configurations matching design.md section 6.1.
///
/// All motion uses spring physics (no duration-based easing).
/// Use with [AnimationController.animateWith] via [createSimulation].
abstract final class ObeliskSprings {
  /// Buttons, quick feedback.
  static final snappy = SpringDescription(mass: 1, stiffness: 400, damping: 38);

  /// Content transitions, general movement.
  static final smooth = SpringDescription(mass: 1, stiffness: 200, damping: 30);

  /// Entries, reveals, fade-ins.
  static final gentle = SpringDescription(mass: 1, stiffness: 150, damping: 28);

  /// Sheet transitions, glass morphing.
  static final liquid = SpringDescription(
    mass: 1.1,
    stiffness: 180,
    damping: 28,
  );

  /// Micro-interactions, immediate response.
  static final quick = SpringDescription(mass: 1, stiffness: 500, damping: 40);

  /// Floating buttons entering view.
  static final floatingEntry = SpringDescription(
    mass: 0.8,
    stiffness: 250,
    damping: 32,
  );

  /// Tab content horizontal transitions.
  static final tabSwipe = SpringDescription(
    mass: 1,
    stiffness: 300,
    damping: 34,
  );

  /// Map pin drop (low damping = bounce).
  static final pinDrop = SpringDescription(
    mass: 1,
    stiffness: 300,
    damping: 20,
  );

  /// Creates a [SpringSimulation] from [spring] animating from [start] to [end].
  static SpringSimulation createSimulation(
    SpringDescription spring, {
    double start = 0,
    double end = 1,
    double velocity = 0,
  }) {
    return SpringSimulation(spring, start, end, velocity);
  }
}
